src/AutorestSwift/View Models/OperationViewModel.swift

```diff
@@ -108,8 +102,26 @@ struct OperationViewModel {
         // TODO: only support max 1 request and  max 1 response for now
         self.request = requests.first
         let response = responses.first
+
+        // All required parameters from the signature paramters from the schema
+        // and the signature parameters from the Request are stored in signatureParams.
+        // All the optional parameters will be in optionalParams
+        var signatureParams = [ParameterViewModel]()
+        var optionalParams = [ParameterViewModel]()
+        for param in operation.signatureParameters ?? [] {
+            if param.required ?? false {
+                signatureParams.append(ParameterViewModel(from: param))
+            } else {
+                optionalParams.append(ParameterViewModel(from: param))
+            }
+        }
```

Comment:
So the logic here isn't really whether the parameter is required or not. It's anything that isn't path or body encoded (i.e. query or header params), which you could get with something like 

```Swift
let optionsParams = operation.signatureParameters.filter { param in
  guard let location = param.protocol.http?.in else { return false }
  let allowed = [.header, .query]
  return allowed.contains(location)
}
```

However, I still think it would be good to have an assert to validate our assumption that anything in options should in fact be optional.

```Swift
for param in optionsParams {
  assert(param.required == false, "Unexpectedly found a required 'option'... \(param.name)")
}
```

Reply:
Likewise, signature params *should* be anything path encoded first, followed by a body parameter, if there is one.

Reply:
Changed per feedback.